Games/Solitaire/Card.h

```diff
@@ -0,0 +1,139 @@
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <LibC/ctype.h>
+#include <LibCore/Object.h>
+#include <LibGUI/Painter.h>
+#include <LibGfx/Bitmap.h>
+#include <LibGfx/CharacterBitmap.h>
+#include <LibGfx/Rect.h>
+
+#define CARD_WIDTH 80
+#define CARD_HEIGHT 100
+#define CARD_COUNT 13
+
+static const NonnullRefPtr<Gfx::CharacterBitmap> s_diamond = Gfx::CharacterBitmap::create_from_ascii(
+    "    #    "
+    "   ###   "
+    "  #####  "
+    " ####### "
+    "#########"
+    " ####### "
+    "  #####  "
+    "   ###   "
+    "    #    ",
+    9, 9);
+
+static const NonnullRefPtr<Gfx::CharacterBitmap> s_heart = Gfx::CharacterBitmap::create_from_ascii(
+    "  #   #  "
+    " ### ### "
+    "#########"
+    "#########"
+    "#########"
+    " ####### "
+    "  #####  "
+    "   ###   "
+    "    #    ",
+    9, 9);
+
+static const NonnullRefPtr<Gfx::CharacterBitmap> s_spade = Gfx::CharacterBitmap::create_from_ascii(
+    "    #    "
+    "   ###   "
+    "  #####  "
+    " ####### "
+    "#########"
+    "#########"
+    " ## # ## "
+    "   ###   "
+    "   ###   ",
+    9, 9);
+
+static const NonnullRefPtr<Gfx::CharacterBitmap> s_club = Gfx::CharacterBitmap::create_from_ascii(
+    "    ###    "
+    "   #####   "
+    "   #####   "
+    " ## ### ## "
+    "###########"
+    "###########"
+    "#### # ####"
+    " ## ### ## "
+    "    ###    ",
+    11, 9);
+
+static RefPtr<Gfx::Bitmap> s_background;
+
+class Card final : public Core::Object {
+    C_OBJECT(Card)
+public:
+    enum Type {
+        Clubs,
+        Diamonds,
+        Hearts,
+        Spades,
+        __Count
+    };
+
+    virtual ~Card() override;
+
+    Gfx::Rect& rect() { return m_rect; }
+    Gfx::Point position() { return m_rect.location(); }
+    Gfx::Point& old_positon() { return m_old_position; }
+    uint8_t value() { return m_value; };
+    Type type() { return m_type; }
+
+    bool is_old_position_valid() const { return m_old_position_valid; }
+    bool is_moving() const { return m_moving; }
+    bool is_upside_down() const { return m_upside_down; }
+    bool is_red() const { return m_red; }
+
+    void set_position(const Gfx::Point p) { m_rect.set_location(p); }
+    void set_moving(bool moving) { m_moving = moving; }
+    void set_upside_down(bool upside_down) { m_upside_down = upside_down; }
+
+    void move(int dx, int dy) { m_rect.move_by(dx, dy); }
+    void save_old_position();
+
+    void draw(GUI::Painter&);
+    void clear(GUI::Painter&, const Color& background_color);
+    void draw_complete(GUI::Painter&, const Color& background_color);
+
+private:
+    explicit Card(Type type, uint8_t value);
+
+    Gfx::Color color() const { return m_red ? Color::Red : Color::Black; }
+
+    Gfx::Rect m_rect;
+    NonnullRefPtr<Gfx::Bitmap> m_front;
+    Gfx::Point m_old_position = Gfx::Point();
```

Comment:
No need to initialize this manually, the default constructor will always run.
```suggestion
    Gfx::Point m_old_position;
```